### PR TITLE
Fixes Issue #173 Part B

### PR DESCRIPTION
**Pull request by jamesneb**
Hey guys,

Drawing backgrounds was a more difficult exercise than orienting text. This is my first attempt - if it is not up to snuff, please let me know.

I'll go ahead and put the ``` if false ``` condition back in before we merge.

### CODE REVIEW

File: map_gui/src/render/road.rs [hunk 1]
```diff
@@ -92,28 +92,30 @@ impl DrawRoad {
                         .zoomed_road_surface(LaneType::Driving, r.get_rank())
                 };
                 // TODO: find a good way to draw an appropriate background
-                if false {
```
Comment (dabreegster) [inline map_gui/src/render/road.rs:95]: Trick for the future: you could change this to `if true` for a simpler diff / way to finalize before merging

File: map_model/src/objects/road.rs [hunk 2]
```diff
@@ -269,6 +269,12 @@ impl Road {
             .min_by_key(|(idx, _)| (our_idx - (*idx as isize)).abs())
             .map(|(_, l)| l)
     }
+    /// This function returns the lane closest to the center from the left hand side of the street
+    pub fn get_lane_underneath_label(&self) -> LaneType {
+        let size = self.lanes.len() - 1;
+        let lane_type_under_label: LaneType = self.lanes[size / 2].lane_type;
```
Comment (dabreegster) [inline map_model/src/objects/road.rs:275]: To match the name of the method, we could return the lane itself, and the caller could then lookup the lane type. We might have some future need to do something similar! The return type would be `&Lane` (a borrow of a Lane), and you could just do `&self.lanes[size / 2]`

File: map_model/src/objects/road.rs [hunk 3]
```diff
@@ -269,6 +269,12 @@ impl Road {
             .min_by_key(|(idx, _)| (our_idx - (*idx as isize)).abs())
             .map(|(_, l)| l)
     }
+    /// This function returns the lane closest to the center from the left hand side of the street
+    pub fn get_lane_underneath_label(&self) -> LaneType {
```
Comment (dabreegster) [inline map_model/src/objects/road.rs:273]: The `map_model` layer of code shouldn't really "know" about rendering. So how about we named this `get_middle_lane` or `get_center_lane` instead? Defining what the middle/center means is a bit vague, but the docstring is great

File: widgetry/src/text.rs [hunk 4]
```diff
@@ -540,16 +540,26 @@ impl TextSpan {
         assets: &A,
         path: &PolyLine,
         scale: f64,
+        color: Color,
```
Comment (dabreegster) [inline widgetry/src/text.rs:543]: Rename `bg_color`

File: widgetry/src/text.rs [hunk 5]
```diff
@@ -540,16 +540,26 @@ impl TextSpan {
...
         let tolerance = svg::HIGH_QUALITY;
 
         // Just set a sufficiently large view box
         let mut svg = r##"<svg width="9999" height="9999" viewBox="0 0 9999 9999" xmlns="http://www.w3.org/2000/svg">"##.to_string();
-
+        let height = (Text::from(&self.text).dims(assets).height * scale);
+        let width = (Text::from(&self.text).dims(assets).width * scale);
+        write!(
+            &mut svg,
+            r##"<rect x="0" y="0" width="{}" height="{}" fill="{}" />"##,
```
Comment (dabreegster) [inline widgetry/src/text.rs:554]: I don't think this approach works. :(

It adds an axis-oriented (no rotation) rectangle based on the text's width & height if we drew it oriented normally, and placed at 0,0. But then below we draw the text translated and oriented to follow a curve. I changed the background color to cyan just to understand what's going on, and didn't see the rectangle under the labels. Instead it's at the top-left corner of the map:

Screenshot from 2022-04-10 21-40-36

I started playing with other ideas to deal with this, like setting `fill="cyan"` on the `<path id="txtpath"` part, but it doesn't do anything nice:
Screenshot from 2022-04-10 21-41-52
I wouldn't expect it to either...

Thinking about this problem a bit more, sometimes the label straddles two lanes, which might be different colors, like this:
Screenshot from 2022-04-10 21-44-48
Even if we calculated a polygon that follows the curved text, picking one of the lane's colors and putting a background wouldn't really work. Really what we want to do here is...

1) Not draw the dashed center line underneath the label
2) Rethink the color of the label, so it'll still show up reasonably clear on these different lane colors

Not sure how to do either easily. :\

**Review by dabreegster — COMMENTED** [review]
Sorry for the super slow review; I've been trying to catch up on work after a vacation. Thanks for working on this!

... But I might've misled you with how easy fixing this would be. I'm a bit stuck what to do.

File: widgetry/src/text.rs (hunk 5 above)
Comment (Zireael07) [reply widgetry/src/text.rs:554]: Looking at the pics, a black label might work? Alternately, what I did with my space game where I had all sorts of planet/nebula colors was to draw a background behind it (gray) and then black text. This ensured the labels are always readable regardless of what crazy color it happened to be on.

Comment (jamesneb) [reply widgetry/src/text.rs:554]: Hey guys,

Thanks for the review :). 

1) I will do this
2) Does SVG allow a stroke color? We could do black stroke with white text

**Pull request closed without merging**